### PR TITLE
*(both): update dep of etcd, go-mysql and others

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4418

### What is changed and how it works?

main purpose is 
- update go-mysql to fix https://github.com/go-mysql-org/go-mysql/issues/677 .
- updating tidb-tools to fix Upstream Switch CI broken

Updating tidb-tools need updating etcd, and further need update many places

### Check List 

Tests 

 - Integration test (dm/tests/adjust_gtid/data/db2.increment.sql)

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -73,35 +73,35 @@ require (
...
-	sigs.k8s.io/yaml v1.2.0 // indirect
 	upper.io/db.v3 v3.7.1+incompatible
 )
 
-replace (
-	// Fix CVE-2020-26160.
-	github.com/dgrijalva/jwt-go v3.2.0+incompatible => github.com/golang-jwt/jwt v3.2.1+incompatible
-	// cloud.google.com/go/storage will upgrade grpc to v1.40.0
-	// we need keep the replacement until go.etcd.io supports the higher version of grpc.
-	google.golang.org/grpc v1.40.0 => google.golang.org/grpc v1.29.1
```
Comment (amyangfei) [inline go.mod:106]: Why we need to pin grpc to `v1.29.1` before this PR

**Review by amyangfei — COMMENTED** [review]
Which release branch will be cherry-picked to?

**Review by lichunzhu — APPROVED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (lance6716) [reply go.mod:106]: introduced in https://github.com/pingcap/tiflow/pull/2833 @maxshuang PTAL

File: go.mod [hunk 2]
```diff
@@ -73,35 +73,35 @@ require (
...
 	github.com/vmihailenco/msgpack/v5 v5.3.5
 	github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
 	github.com/xitongsys/parquet-go v1.6.0 // indirect
-	go.etcd.io/etcd v0.5.0-alpha.5.0.20210512015243-d19fbe541bf9
+	go.etcd.io/etcd/api/v3 v3.5.2
+	go.etcd.io/etcd/client/pkg/v3 v3.5.2
+	go.etcd.io/etcd/client/v3 v3.5.2
+	go.etcd.io/etcd/pkg/v3 v3.5.2
+	go.etcd.io/etcd/server/v3 v3.5.2
+	go.etcd.io/etcd/tests/v3 v3.5.2
```
Comment (overvenus) [inline go.mod:89]: Which version does PD use?

**Comment by lance6716** [comment]
> Which release branch will be cherry-picked to?

https://github.com/pingcap/tidb/pull/32457 TiDB doesn't cherry-pick

File: go.mod (hunk 2 above)
Comment (lance6716) [reply go.mod:89]: @disksing has updated all active repos but forgotten us 😭

Comment (disksing) [reply go.mod:89]: pd/server uses old version but pd/client uses v3.5.2

File: go.mod (hunk 1 above)
Comment (disksing) [reply go.mod:106]: grpc has directory changes, old etcd cannot use newer grpc.

**Comment by lance6716** [comment]
/run-all-tests

all tests (except for CDC's broken CI) should have passed now, PTAL @amyangfei @overvenus @hi-rustin

**Comment by lance6716** [comment]
CDC's CI have benn passed before, I think it's OK to review now

https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_kafka_integration_test/detail/cdc_ghpr_kafka_integration_test/9387/pipeline/
https://ci2.pingcap.net/blue/organizations/jenkins/cdc_ghpr_integration_test/detail/cdc_ghpr_integration_test/10126/pipeline

**Review by amyangfei — APPROVED** [review]

**Comment by lance6716** [comment]
waiting https://github.com/pingcap/tiflow/pull/4783 to fix CI

**Pull request merged**